File: src/bot/commands/speed.ts
import type { Message } from 'discord.js-selfbot-v13';
import { getVideoStreamer } from '../../stream/video-streamer.js';

export async function speedCommand(message: Message, args: string[]): Promise<void> {
  if (!message.guild) {
    await message.channel.send('❌ This command can only be used in a server');
    return;
  }

  const guildId = message.guild.id;
  const videoStreamer = getVideoStreamer();
  const session = videoStreamer.getSession(guildId);

  if (!session) {
    await message.channel.send('❌ Nothing is currently playing');
    return;
  }

  const speedArg = args[0];
  
  // If no argument, show current speed
  if (!speedArg) {
    const currentSpeed = videoStreamer.getSpeed(guildId);
    await message.channel.send(`⏩ Current playback speed: **${currentSpeed}x**\n\nUsage: \`!speed <multiplier>\`\nExamples: \`!speed 1.5\`, \`!speed 2\`, \`!speed 0.75\`\nRange: 0.5x - 3x`);
    return;
  }

  // Parse speed value
  let speed = parseFloat(speedArg.replace('x', ''));
  
  if (isNaN(speed)) {
    await message.channel.send('❌ Invalid speed. Use a number like `1.5` or `2x`');
    return;
  }

  // Clamp speed
  speed = Math.max(0.5, Math.min(3, speed));

  const statusMsg = await message.channel.send(`⏩ Setting playback speed to ${speed}x...`);

  const success = await videoStreamer.setSpeed(guildId, speed);

  if (success) {
    const speedEmoji = speed > 1 ? '⏩' : speed < 1 ? '⏪' : '▶️';
    await statusMsg.edit(`${speedEmoji} Playback speed set to **${speed}x**`);
  } else {
    await statusMsg.edit('❌ Failed to change playback speed');
  }
}

File: selfbot/commands/speed.py
"""Speed command via IPC."""

from __future__ import annotations

from typing import TYPE_CHECKING
import discord

from ..ipc import ipc

if TYPE_CHECKING:
    from ..client import SchroStreamBot


async def speed_command(bot: SchroStreamBot, message: discord.Message, args: list[str]) -> None:
    if not message.guild:
        await message.channel.send("❌ This command can only be used in a server")
        return

    guild_id = str(message.guild.id)

    # No args — show current speed
    if not args:
        result = await ipc.get_now_playing(guild_id)
        if result.get("success"):
            spd = result.get("speed", 1.0)
            await message.channel.send(
                f"⏩ Current playback speed: **{spd}x**\n\n"
                "Usage: `!speed <multiplier>`\n"
                "Examples: `!speed 1.5`, `!speed 2`, `!speed 0.75`\n"
                "Range: 0.5x - 3x"
            )
        else:
            await message.channel.send("❌ Nothing is currently playing")
        return

    speed_str = args[0].replace("x", "")
    try:
        speed = float(speed_str)
    except ValueError:
        await message.channel.send("❌ Invalid speed. Use a number like `1.5` or `2x`")
        return

    speed = max(0.5, min(3.0, speed))

    status_msg = await message.channel.send(f"⏩ Setting playback speed to {speed}x...")

    result = await ipc.control_speed(guild_id, speed)

    if result.get("success"):
        emoji = "⏩" if speed > 1 else "⏪" if speed < 1 else "▶️"
        await status_msg.edit(content=f"{emoji} Playback speed set to **{speed}x**")
    else:
        await status_msg.edit(content="❌ Failed to change playback speed")
